### ISSUE TITLE
v157 scid is no longer posted for shipping_estimator.

### ISSUE THREAD

**Issue by mc12345678** [zencart/zencart#4838]
Commit https://github.com/zencart/zencart/commit/74b97cbdcc616f4d9f58ea8823f14a1957d0ab8a eliminated a PHP notice by validating that `$selected_shipping` was an `array` before attempting to reference it as one.

The change though removed the `echo` command before calling `zen_draw_hidden_field`. As a result the field `'scid'` is not pushed to the browser in later processing (`includes/modules/shipping_estimator.php`).

This appears to cause an absence of setting `$_SESSION['cart_sid']`, This may also cause `$method` and `$module` to not be set and at least lose out on information expected to made available for the shipping_estimator.

Fix is perceived to be to change: https://github.com/zencart/zencart/blob/742b837868a2b128df96b6635c413c8187972679/includes/templates/template_default/templates/tpl_modules_shipping_estimator.php#L13-L15

````diff
<?php if (is_array($selected_shipping)) {
-    zen_draw_hidden_field('scid', $selected_shipping['id']);
+    echo zen_draw_hidden_field('scid', $selected_shipping['id']);
} ?>
````

(Removal in red, addition in green)

As far as "issues" that it causes? I haven't put myself in a position yet to figure that out. I just identified the above absent information from my understanding of the Zen Cart code, the change that was made and seeing that there is at least something in the files depending on the hidden variable. Would have to follow the code further and possibly run some tests to see how much of an impact this has on overall operation.

Based on the github fileset, it is applicable to all of ZC Version 1.5.7 and its sub-versions and is seen to remain present on version 1.5.8 as well. It appears that this may affect shipping modules that have multiple methods. Perhaps more could be looked into that. This is an initial report so that I may continue on with the task I need to complete.

**Comment by proseLA** [comment]
@torvista ???

**Comment by scottcwilson** [comment]
Fixed in PR #4884
